src/assemble_workflow/bundle_rpm.py

```diff
@@ -89,14 +92,27 @@ def build(self, name: str, dest: str, archive_path: str) -> None:
         shutil.move(f"{min_bin_env_path}.backup", min_bin_env_path)
         shutil.move(min_dest_path, min_source_path)
 
+        # Append correct value substitute to the spec file
+        architecture_alt_rpm = {
+            "x64": "x86_64",
+            "arm64": "aarch64",
+        }
+        replaces = ('##VERSION##', build_kls.version), ('##ARCHITECTURE_ALT##', architecture_alt_rpm[build_kls.architecture])
+
+        with open(os.path.join(ext_dest, f"{self.filename}.rpm.spec"), 'r') as fp:
+            spec_lines = fp.read()
+
+        with open(os.path.join(ext_dest, f"{self.filename}.rpm.final.spec"), 'w') as fp:
+            fp.write(reduce(lambda a, key_value: a.replace(*key_value), replaces, spec_lines))
+
         # Run bundle rpmbuild
         bundle_cmd = " ".join(
             [
                 'rpmbuild',
                 '-bb',
                 '--define',
                 f"'_topdir {ext_dest}'",
-                f"{self.filename}.rpm.spec",
+                f"{self.filename}.rpm.final.spec",
```

Comment:
As is, the .spec file is not a real .spec (it includes variables such as `###VERSION###` that's not valid syntax for a spec file. So it should be called `.spec.template`, and its output `.spec`. 

But I'd prefer it if you used the built-in variable substitution mechanism and thus .spec was valid in both the source and final version. Since the .spec file is checked in, the version that is generated should be placed in a destination/target folder under the same name.

Reply:
This is a valid call out, I will create an issue to improve this later.

Reply:
I'd rather fix this here. It's a trivial change and the issue will just sit there forever.

Reply:
Fixed now.